src/render.c

```diff
@@ -70,7 +70,9 @@ void render_con(Con *con, bool render_fullscreen, bool already_inset) {
         }
         inset.height = 0;
         if (con->deco_rect.width != 0 && con->deco_rect.height != 0) {
-            con->deco_rect = rect_add(con->deco_rect, inset);
+            if (con_is_leaf(con)) {
+                con->deco_rect = rect_add(con->deco_rect, inset);
```

Comment:
This time around instead of me saying that I think this isn't correct, I'll reverse the roles: can you explain why you think this change is correct?

Reply:
It probably isn't correct, however, I think it's _more_ correct than what's already there. It fixes the problem I was having, and, as far as I can tell, doesn't cause any other problems.

Reply:
> It probably isn't correct, however, I think it's more correct than what's already there.

Replacing (supposedly) broken code with other broken code that isn't understood doesn't sound like a good plan to me. Right now, things seem to work fine for the vast majority of people. I'm not going to risk fixing it for you, but perhaps breaking it for many others with a patch we cannot explain.

I'm happy to consider a PR where you can reason over the change, but I'm not going to spend time thinking over guesses.